### PR TITLE
Avoiding to compile parts of statements with preprocessor conditionals.

### PR DESCRIPTION
**Pull request by flaviommedeiros**
As discussed in pull request #465, I'm submitting a new pull request with the goal of removing all conditional directives of the following pattern:

#ifdef A
if (condition_1)
#else
if (condition_2)
#endif
{
// code
}

I have used a tool that searches for this pattern and I have refactored them manually. We are able to detect other patterns of conditional directives that split up parts of statements to refactor manually (if it contributes to OpenSSL).

Let me know if there are other patterns that are important to refactor in order to improve code understanding and so on.

### CODE REVIEW

File: apps/req.c [hunk 1]
```diff
@@ -1134,24 +1134,28 @@ static int auto_info(X509_REQ *req, STACK_OF(CONF_VALUE) *dn_sk,
...
+                    || (*p == os_toascii['.']));
 #endif
+            if (spec_char){
                 p++;
                 if (*p)
                     type = p;
                 break;
             }
+        }
+        int plus_char;
```
Comment (richsalz) [inline apps/req.c]: still not a C feature.  Needs to be inside a block.  Suggest moving all variables to the start of the function just "to be safe" and avoid these problems.

**Comment by flaviommedeiros** [comment]
I moved all variables to the beginning of the functions.

File: apps/req.c [hunk 2]
```diff
@@ -1134,24 +1134,26 @@ static int auto_info(X509_REQ *req, STACK_OF(CONF_VALUE) *dn_sk,
...
+            if (spec_char){
                 p++;
                 if (*p)
                     type = p;
                 break;
             }
+        }
 #ifndef CHARSET_EBCDIC
-        if (*p == '+')
+        plus_char = (*p == ‘+’);
```
Comment (kroeckx) [inline apps/req.c]: Those are the wrong quotes.

**Comment by flaviommedeiros** [comment]
Any other recommendation regarding the new pull request #480, @kroeckx and @richsalz?

**Comment by richsalz** [comment]
We use space before the curly brace, not "){" but that's trivial to fix.

**Pull request closed without merging**